### PR TITLE
Add database backup workflow for aks

### PR DESCRIPTION
**Pull request by johnake**
### Context

Add database backup workflow for aks

### Changes proposed in this pull request

Include a summary of the change.

### Guidance to review

Include any useful information needed to review this change.
Include any dependencies that are required for this change.

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by johnake** [comment]
[Add database-backup workflow for aks](https://trello.com/c/D0tTlDNd/737-add-database-backup-workflow-for-aks)

File: .github/workflows/actions/database-backup/action.yml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+        if [ -z "$APP_ENVIRONMENT" ]; then
+            echo "::error ::Failed to extract app_environment from $TFVARS"
+            exit 1
+        fi
+
+        echo ::set-output name=cluster::$CLUSTER
+        echo ::set-output name=namespace::$NAMESPACE
+        echo ::set-output name=cluster::$CLUSTER
+        echo ::set-output name=app_environment::$APP_ENVIRONMENT
```
Comment (temitope777) [inline .github/workflows/actions/database-backup/action.yml:60]: set-output is deprecated so use $GITHUB_OUTPUT here

File: .github/workflows/actions/database-backup/action.yml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      env:
+        TFVARS: ${{ inputs.tfvars }}
+      working-directory: terraform/aks
+
+    - uses: Azure/login@v1
+      with:
+        creds: ${{ inputs.azure_credentials }}
+
+    - uses: DfE-Digital/keyvault-yaml-secret@v1
+      id: keyvault-yaml-secret
```
Comment (temitope777) [inline .github/workflows/actions/database-backup/action.yml:71]: we are moving away from yaml secret. use key vault values
see example in https://github.com/DFE-Digital/schools-experience/blob/master/.github/workflows/build.yml#L77

File: .github/workflows/actions/database-backup/action.yml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+description: backs up AKS database to Azure Storage
+
+inputs:
+  environment:
+    description: "The name of the environment"
+    required: true
+  azure_credentials:
+    description: "JSON object containing a service principal that can read from Azure Key Vault"
+    required: true
+  restoreToProductionDataEnv:
```
Comment (temitope777) [inline .github/workflows/actions/database-backup/action.yml:11]: i dont think you are using this input

File: .github/workflows/database-backup.yml [hunk 4]
```diff
@@ -2,6 +2,12 @@ name: Backup Database to Azure Storage
 
 on:
   workflow_dispatch:
+    inputs:
```
Comment (temitope777) [inline .github/workflows/database-backup.yml:5]: are you using this file?

File: .github/workflows/database-backup.yml [hunk 5]
```diff
@@ -23,6 +29,7 @@ jobs:
         shell: bash
         run: |
           tf_vars_file=terraform/paas/workspace_variables/production.tfvars.json
+          echo "namespace=$(jq -r '.namespace' ${tf_vars_file})" >> $GITHUB_ENV
```
Comment (temitope777) [inline .github/workflows/database-backup.yml:32]: it seems this is backing up paas db but confirm if we still need to backup paas db going forward

File: .github/workflows/aks-db-backup.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+name: Backup AKS Database
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: Environment
+        type: choice
```
Comment (temitope777) [inline .github/workflows/aks-db-backup.yml:8]: do you still need these options when it is scheduled?

File: .github/workflows/database-backup.yml [hunk 7]
```diff
@@ -2,6 +2,12 @@ name: Backup Database to Azure Storage
 
 on:
```
Comment (temitope777) [inline .github/workflows/database-backup.yml:3]: This is backing up paas db. i dont know if we need it after migration

**Review by temitope777 — APPROVED** [review]

**Pull request merged**